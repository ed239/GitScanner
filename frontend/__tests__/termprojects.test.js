import { render, screen,fireEvent } from "@testing-library/react";
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import TermProjects from "../src/pages/termProjects";

 
describe("TermProjects", () => {
    let input;
    test(" TermProjects ", () => {
        const sheetdatas=[['name','1','https://www.baidu.com/']];
        render(<TermProjects sheetdata={sheetdatas} />); 
        sheetdatas.forEach((item) => {
          expect(screen.getByText(item[0])).toBeInTheDocument(); // 确保每个数据项都在文档中存在
          expect(screen.getByText(item[1])).toBeInTheDocument(); // 确保每个数据项都在文档中存在
          expect(screen.getByText(item[2])).toBeInTheDocument(); // 确保每个数据项都在文档中存在
        });
    }); 
    test(" TermProjects Copy Link", () => {
      const sheetdatas=[['name','1','https://www.baidu.com/']];
      render(<TermProjects sheetdata={sheetdatas} />); 
      const CopyLinkBut = screen.getByText("Copy Link"); 
      expect(CopyLinkBut).toBeInTheDocument();
      //fireEvent.click(CopyLinkBut);
  }); 
});
