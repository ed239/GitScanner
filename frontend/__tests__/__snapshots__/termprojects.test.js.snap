// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TermProjects should render correctly with a non-empty array 1`] = `undefined`;

exports[`heading component renders a heading 1`] = `undefined`;
